src/ReactiveSocket.cpp

```diff
@@ -151,6 +157,14 @@ bool ReactiveSocket::createResponder(
       handler_->handleFireAndForgetRequest(std::move(frame.data_));
       break;
     }
+    case FrameType::METADATA_PUSH: {
```

Comment:
Simplest way is in ReactiveSocket::ReactiveSocket, passing it into the ConnectionAutomation constructor. I don't think you have a choice since ConnectionAutomaton::onNext won't call this block for streamId 0.